src/Equations/Fluid/ParallelHeatLossTerm.cpp

```diff
@@ -0,0 +1,245 @@
+            minZ = currentZ;          
+            minIndex = i;             
+        }
+    }
+
+    // Check if a valid minimum Z was found and set the ion quantities
+    if (minIndex != -1) {
+        this-> m_i = ions->GetIonSpeciesMass(minIndex); 
+        this-> Z = minZ;  
+                                     
+    }
+}
+
+/**
+ * Destructor
+ */
+ParallelHeatLossTerm::~ParallelHeatLossTerm(){
+    Deallocate();
+
+}           
+
+
+/**
+ * Called after the grid is rebuilt; (re)allocates memory
+ * for all quantities
+ */
+bool ParallelHeatLossTerm::GridRebuilt(){
+
+    this->DiagonalComplexTerm::GridRebuilt();
+    Deallocate();
+    this->HeatLoss = new real_t[nr];
+    
+    return true;
+}
+
+
+/**
+* Find current radial point of last closed flux surface
+* and find psi_edge_t0 if not provided in input
+*/
+void ParallelHeatLossTerm::FindRadiusOfLCFS(){
+    
+    if(this->userGivenPsiEdge_t0 == 0) {
+        
+        // Extrapolate psi from last two ir:s to find the approximate edge value at t=0
+        real_t Rlow = this->rGrid->GetR(nr-2);
+        real_t Rhigh = this->rGrid->GetR(nr-1);
+        real_t Psilow = unknowns->GetUnknownInitialData(id_psi)[nr-2];
+        real_t Psihigh = unknowns->GetUnknownInitialData(id_psi)[nr-1];
+        real_t slope = (Psihigh - Psilow) / (Rhigh - Rlow);
+        real_t Redge = this->rGrid->GetR_f(nr);
+        
+        this->psi_edge_t0 = Psihigh + slope * (Redge - Rhigh); 
+    }
+    
+    // Find if the sign of PsiDiff() should be +1 or -1
+    if(this->signFixed == false) { 
+        real_t psi_mid_init = unknowns->GetUnknownInitialData(id_psi)[0];
+        real_t psi_edge_init = unknowns->GetUnknownInitialData(id_psi)[nr-1];
+        if(psi_mid_init-psi_edge_init > 0) { this->sign = 1; }
+        this->signFixed = true;
+    }
+    
+    bool exists = false;
+    for (len_t ir = 0; ir < nr; ir++){
+        if(this->sign * PsiDiff(ir) >= 0){
+            this->ir_LCFS = (int_t)ir; // Last ir to pass critera is ir_LCFS
+            exists = true;
+        }
+    }
+    if ( exists == false ) { this->ir_LCFS = -1; } // No flux surface is closed
+}
+
+
+/**
+* Difference between psi_p at ir+1/2 and the edge psi_p
+*/
+real_t ParallelHeatLossTerm::PsiDiff(len_t ir){
+    // Interpolate (extrapolate for ir=0) psi for estimate of psi at inner 
+    // radial grid cell wall of current grid cell
+    real_t psi_f = InterpolatePsi(ir); 
+    return psi_f - this->psi_edge_t0;
+}
+
+
+/**
+* Interpolate flux between ir:s to get psi at radial grid cell walls 
+* (Doesn't work for nr=1, add warning / if...else... with old condition in PsiDiff?)
+*/
+real_t ParallelHeatLossTerm::InterpolatePsi(len_t jr){
+
+    // jr index for grid cell walls
+    len_t ir_j; // Helper index, lowest ir for which to find psi 
+    if (jr == 0) {
+        ir_j = 0; // Extrapolate for jr=0
+    } else {
+        ir_j = jr-1;
+    }
+    
+    // Need special solution if nr=1(0)?
```

Comment:
Should we handle the case ``nr=1``? For example by adding the following to the constructor?
```c++
if (nr == 1)
    throw EquationException("ParallelHeatLossTerm: The halo region heat loss term requires nr > 1.");
```